From a997978841f388063d97e5bf1cc756d28f290547 Mon Sep 17 00:00:00 2001
From: Sean Mackrory <mackrorysd@apache.org>
Date: Wed, 15 Feb 2017 15:02:48 -0700
Subject: [PATCH 2277/2848] HADOOP-14085. s3guard: drop unnecessary type
 assertion and cast.

(cherry picked from commit 94287ce70af7b71ef0a59330b83a61561fcfb26d)

Change-Id: Ia9a18c0d69b48a3ec4145c03a29c0da0bc196602
---
 .../s3guard/PathMetadataDynamoDBTranslation.java   |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/s3guard/PathMetadataDynamoDBTranslation.java b/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/s3guard/PathMetadataDynamoDBTranslation.java
index a8d0974..3445650 100644
--- a/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/s3guard/PathMetadataDynamoDBTranslation.java
+++ b/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/s3guard/PathMetadataDynamoDBTranslation.java
@@ -144,8 +144,7 @@ static PathMetadata itemToPathMetadata(Item item, String username)
    */
   static Item pathMetadataToItem(PathMetadata meta) {
     Preconditions.checkNotNull(meta);
-    assert meta.getFileStatus() instanceof S3AFileStatus;
-    final S3AFileStatus status = (S3AFileStatus) meta.getFileStatus();
+    final FileStatus status = meta.getFileStatus();
     final Item item = new Item().withPrimaryKey(pathToKey(status.getPath()));
     if (status.isDirectory()) {
       item.withBoolean(IS_DIR, true);
-- 
1.7.9.5

